core-chart/templates/postcreatehooks/install-status-addon.yaml

```diff
@@ -0,0 +1,123 @@
+{{- if .Values.InstallPCHs }}
+apiVersion: tenancy.kflex.kubestellar.org/v1alpha1
+kind: PostCreateHook
+metadata:
+  name: install-status-addon
+  labels:
+    kflex.kubestellar.io/cptype: its
+spec:
+  templates:
+  - apiVersion: batch/v1
+    kind: Job
+    metadata:
+      name: "{{"{{.HookName}}"}}"
+    spec:
+      template:
+        spec:
+          initContainers:
+          - name: wait-for-kubeconfig
+            image: quay.io/kubestellar/kubectl:{{.Values.KUBECTL_VERSION}}
+            command:
+              - sh
+              - -c
+              - |
+                echo "Waiting for kubeconfig file /etc/kube/{{.ITSkubeconfig}} to exist and be non-empty..."
+                until [ -s /etc/kube/{{.ITSkubeconfig}} ]; do
+                  echo "Kubeconfig file not ready yet, waiting..."
+                  sleep 2
+                done
+                echo "Kubeconfig file is ready"
+            volumeMounts:
+            - name: kubeconfig
+              mountPath: "/etc/kube"
+              readOnly: true
+
+          - name: wait-for-ocm-crds
+            image: quay.io/kubestellar/kubectl:{{.Values.KUBECTL_VERSION}}
+            command:
+              - sh
+              - -c
+              - |
+                echo "Waiting for OCM CRDs to be established..."
+                # Wait for essential OCM CRDs needed by the Status Add-on
+                # First check if clusteradm has created any OCM CRDs
+                echo "Checking if OCM CRDs exist..."
+                max_attempts=60
+                attempt=0
+                while [ $attempt -lt $max_attempts ]; do
+                  if kubectl get crd placements.cluster.open-cluster-management.io >/dev/null 2>&1; then
```

Comment:
Is it enough to only wait for one of them to be created? Are we guaranteed that this one is created last?